core/common/src/main/java/alluxio/util/EnvironmentUtils.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.util;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Utilities to detect environment Alluxio is running in.
+ */
+@ThreadSafe
+public final class EnvironmentUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(EnvironmentUtils.class);
+
+  /**
+   * Utility to detect the docker deployment environment.
+   *
+   * @return true, if running on docker
+   */
+  @SuppressFBWarnings("DMI_HARDCODED_ABSOLUTE_FILENAME")
+  public static boolean isDocker() throws IOException {
```

Comment:
do we want these status methods to throw an exception?

Reply:
we catch all exceptions in the metamaster update check code. i feel that's better than hiding the exception and choosing a default